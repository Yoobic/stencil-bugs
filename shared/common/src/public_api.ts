export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';