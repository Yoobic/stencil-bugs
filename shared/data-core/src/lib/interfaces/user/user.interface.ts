import { Model } from '../../decorators/model/model.decorator';
import { Searchable } from '../../decorators/searchable/searchable.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { momentTimeZone, FormFieldType, IUser, IUserSettings, ISimpleUser } from '@shared/interfaces';
import { Location } from '../location/location.interface';
import { Tenant } from '../tenant/tenant.interface';
import { usernameValidatorRequired } from '../../validators/username-validator/username-validator';
import { ROLES_CONDITIONS, getGroupsTransform } from '../condition/condition.interface';

import { startCase } from 'lodash-es';

let conditions = {
    isCreate: 'not getAttributeValue("_ect")',
    isUpdate: 'getAttributeValue("_ect")',
    isNotUpdateMultiple: 'not (getAttributeValue("_id") == "multiple")',
    //isTeam: ' isTeam == 1',
    //isNotTeam: ' not (isTeam == 1)',
    isNotSarahMartin: 'not (username == "smartin@yoobic.com")',
    isNotYoobicAdmin: 'not (endsWith(getAttributeValue("username"),"@yoobic.com") == 1 and length("_acl.groups.r") == 0)'
};

export function onUserLocationChange(value, controls, data) {
    if (!value) {
        data.locationRef = null;
    }
}

@Model({
    className: 'User',
    collectionName: 'user',
    fields: ['*'],
    include: ['location', '_tenant']
})

export class User extends IUser {

    @Editable('User', { type: FormFieldType.text, visible: false, forceExport: true, exportOrder: 11 })
    _id?: string;

    @Editable('User', { type: FormFieldType.photo, filterable: false, title: 'PHOTO', columnDefinition: { width: 52 }, exportOrder: 10, sortable: false, allowLibrary: true })
    imageData?: string;

    @Editable('User', { required: true, flex: 100, type: FormFieldType.email, disableAutocomplete: true, condition: conditions.isNotUpdateMultiple, readonly: conditions.isUpdate, asyncValidators: [usernameValidatorRequired], exportOrder: 1 })
    @Searchable('User')
    username: string;

    @Editable('User', { required: false, flex: 100, type: FormFieldType.email, disableAutocomplete: true, exportOrder: 6 })
    @Searchable('User')
    email?: string;

    @Editable('User', { required: true, type: FormFieldType.password, disableAutocomplete: true, condition: conditions.isCreate, filterable: false, sortable: false, exportOrder: 2 })
    password?: string;

    @Editable('User', { required: true, flex: 100, type: FormFieldType.text, exportOrder: 4 })
    @Searchable('User')
    firstName?: string;

    @Editable('User', { flex: 100, required: true, type: FormFieldType.text, exportOrder: 5 })
    @Searchable('User')
    lastName?: string;

    @Editable('User', { required: true, title: 'TENANT', type: FormFieldType.autocomplete, condition: [ROLES_CONDITIONS.isAdmin], collectionName: 'tenants', multiple: false, columnDefinition: { name: 'name' } })
    _tenant: Tenant;

    @Editable('User', { type: FormFieldType.toggle, flex: 100, title: 'TEAM', exportOrder: 12, condition: [conditions.isCreate, ROLES_CONDITIONS.isAdmin], filterableAdvanced: true })
    isTeam?: boolean;

    @Editable('User', { title: 'USERTAGS', type: FormFieldType.autocomplete, tag: true, collectionName: 'user', condition: ROLES_CONDITIONS.isAdmin, multiple: true, icon: 'yo-flag', subQuery: { field: 'ownerRef', values: '_id' }, exportOrder: 13 })
    tags?: Array<string>;

    @Editable('User', { flex: 100, required: true,  type: FormFieldType.autocomplete, exportOrder: 15, condition: [ROLES_CONDITIONS.isAdmin], values: ['ROLEADMIN', 'ROLEEDITOR', 'ROLEVIEWER', 'ROLEMANAGER', 'ROLEFIELD', 'ROLESTOREMANAGER', 'ROLESTORE'], translate: true, filterableAdvanced: true })
    role?: string;

    @Editable('User', { flex: 50, type: FormFieldType.tel, exportOrder: 7, filterableAdvanced: true, advanced: true })
    telephone?: string;

    @Editable('User', { flex: 50, type: FormFieldType.text, exportOrder: 8, filterableAdvanced: true, advanced: true })
    company?: string;

    //@Editable('User', { flex: 100, type: FormFieldType.text, exportOrder: 16, filterableAdvanced: true })
    position?: string;

    @Editable('User', { flex: 50, type: FormFieldType.date, suppressExport: true, filterableAdvanced: true, condition: ROLES_CONDITIONS.isNotTrial, advanced: true })
    dateOfBirth?: string;

    @Editable('User', { flex: 50, type: FormFieldType.autocomplete, values: ['MALE', 'FEMALE'], translate: true, suppressExport: true, filterableAdvanced: true, condition: ROLES_CONDITIONS.isNotTrial, advanced: true })
    gender?: string;

    //@Editable('User', { type: FormFieldType.email, condition: conditions.isNotTeam, exportOrder: 17, filterableAdvanced: true })

    @Editable('User', { type: FormFieldType.address, exportOrder: 9, condition: ROLES_CONDITIONS.isNotTrial, advanced: true })
    address?: any;

    @Editable('User', { title: 'LOCATION', type: FormFieldType.autocomplete, onChange: onUserLocationChange, collectionName: 'locations', deleteOnHidden: false, condition: [ROLES_CONDITIONS.isManager, ROLES_CONDITIONS.isNotTrial, conditions.isNotSarahMartin, conditions.isNotYoobicAdmin], suppressExport: true, clearable: true })
    location?: Location;

    @Editable('User', { type: FormFieldType.text, visible: false, forceExport: true, exportOrder: 14, filterable: false })
    locationRef?: string;

    @Editable('User', {
        required: true, name: '_acl.groups.r', columnDefinition: { name: '_acl?.groups?.r', forceName: true },
        title: 'GROUPS', condition: <any>[ROLES_CONDITIONS.isAdminOrClientAdmin, <any>conditions.isNotSarahMartin, conditions.isNotYoobicAdmin, ROLES_CONDITIONS.isNotTrial],
        type: FormFieldType.autocomplete, allowCustomTag: true, collectionName: 'groups',
        filters: [[{ field: 'isRole', operator: { _id: 'neq' }, value: true }, { field: 'type', operator: { _id: 'nin' }, value: ['plan'] }]], hiddenFields: ['isRole', 'type'],
        idOnly: true, multiple: true, clearable: false, exportOrder: 3, deleteOnHidden: false,
        mapTransform: getGroupsTransform
    })
    _aclGroupsR?: any;

    @Editable('User', { type: FormFieldType.date, name: '_lmt', readonly: true, suppressExport: true, filterableAdvanced: true, advanced: true })
    lastSeen?: any;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: false, filterableAdvanced: true, advanced: true })
    version?: any;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: false, filterableAdvanced: true, advanced: true })
    mobileVersion?: any;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: true, filterableAdvanced: true, advanced: true })
    platform?: any;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: true, filterableAdvanced: true, advanced: true })
    language?: string;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: true, filterable: false, advanced: true })
    device?: string;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: true, filterable: false, advanced: true })
    uuid?: string;

    @Editable('User', { type: FormFieldType.text, readonly: true, suppressExport: true, filterable: false, advanced: true })
    target?: string;

    @Editable('User', { title: 'MAXWIDTH', type: FormFieldType.number, condition: ROLES_CONDITIONS.isAdmin, suppressExport: true, filterable: false })
    photoMaxWidth?: number;

    static getDisplayName(user: any): string {
        if (user) {
            let displayName = user.username;
            if (user.firstName && user.lastName) {
                displayName = startCase(user.firstName.toString().toLowerCase()) + ' ' + startCase(user.lastName.toString().toLowerCase());
            } else if (user.first_name && user.last_name) {
                displayName = startCase(user.first_name.toString().toLowerCase()) + ' ' + startCase(user.last_name.toString().toLowerCase());
            } else if (user._socialIdentity && user._socialIdentity.facebook && user._socialIdentity.facebook.name) {
                displayName = user._socialIdentity.facebook.name;
            } else if (user.email) {
                displayName = user.email;
            } else if (user._socialIdentity && user._socialIdentity.facebook && user._socialIdentity.facebook.email) {
                displayName = user._socialIdentity.facebook.email;
            }
            return displayName;
        }
        return '';
    }

    static getSimpleFields() {
        return ['_id', 'imageData', 'firstName', 'lastName', 'email', 'telephone', 'username'];
    }
}

@Model({ className: 'UserSettings' })
export class UserSettings extends IUserSettings {

    // @Editable('UserSettings', { title: 'Beta', header: 'APPEARANCE', type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })

    @Editable('UserSettings', { type: FormFieldType.toggle, header: 'APPEARANCE', flex: 100, suppressExport: true, filterable: false, sortable: false })
    viewlist: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })
    darkTheme: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })
    useBigFonts: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })
    showScrollbars: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, header: 'NOTIFICATIONS', flex: 100, title: 'DISABLEALLEMAILS', suppressExport: true, filterable: false, sortable: false })
    disableEmailNotifications: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, title: 'DISABLEALLPUSH', suppressExport: true, filterable: false, sortable: false })
    disablePushNotifications: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, title: 'DISABLEALLSMS', suppressExport: true, filterable: false, sortable: false })
    disableSmsNotifications: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })
    sendFinishedEmail: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, header: 'PHOTOS', flex: 100, title: 'DELETEPHOTOSINALBUM', suppressExport: true, filterable: false, sortable: false })
    deletePhotos: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })
    allowPhotoEdit: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, suppressExport: true, filterable: false, sortable: false })
    disablePhotoOrientationAutoFix: boolean;

    @Editable('UserSettings', { title: 'MAXWIDTH', type: FormFieldType.number, suppressExport: true, filterable: false, min: 600, max: 2048 })
    photoMaxWidth?: number;

    @Editable('UserSettings', { type: FormFieldType.toggle, flex: 100, header: 'SERVICES', suppressExport: true, filterable: false, sortable: false })
    orderServicesByDate: boolean;

    @Editable('UserSettings', { type: FormFieldType.toggle, header: 'MISSIONS', flex: 100, suppressExport: true, filterable: false, sortable: false })
    goToBasket: boolean;

    @Editable('UserSettings', { type: FormFieldType.range, min: 1, max: 400, suppressExport: true, filterableAdvanced: true })
    radius: any;

    @Editable('UserSettings', { type: FormFieldType.autocomplete, values: momentTimeZone.tz.names() })
    timezone: any;

    // @Editable('UserSettings', { type: FormFieldType.selectbuttons, flex: 100, suppressExport: true, filterable: false, sortable: false, values: ['YES', 'NO'] })
    // test;
}

@Model({ className: 'SimpleUser' })
export class SimpleUser extends ISimpleUser {
    // export class SimpleUser implements IEntity {
    @Editable('SimpleUser', { type: FormFieldType.text, visible: false })
    _id?: string;

    @Editable('SimpleUser', { visible: true, type: FormFieldType.photo, filterable: false, title: 'PHOTO', columnDefinition: { width: 52 } })
    imageData: string;

    @Editable('SimpleUser', { required: true, type: FormFieldType.email, asyncValidators: [usernameValidatorRequired] })
    @Searchable('SimpleUser') username: string;

    @Editable('SimpleUser', { visible: true, required: true, type: FormFieldType.email })
    @Searchable('SimpleUser') email: string;

    @Editable('SimpleUser', { required: true, type: FormFieldType.password, condition: conditions.isCreate, filterable: false, sortable: false })
    password: string;
}