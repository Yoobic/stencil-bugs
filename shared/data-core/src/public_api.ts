export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
