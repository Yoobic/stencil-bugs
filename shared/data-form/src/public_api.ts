
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';