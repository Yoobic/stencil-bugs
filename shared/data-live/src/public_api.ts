
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';