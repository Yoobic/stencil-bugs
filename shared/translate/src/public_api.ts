export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';