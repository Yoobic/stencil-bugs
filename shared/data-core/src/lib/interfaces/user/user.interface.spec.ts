import { User } from './user.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: User', () => {
    it('should create an entity', () => {
        let entity = new User();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('User');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});