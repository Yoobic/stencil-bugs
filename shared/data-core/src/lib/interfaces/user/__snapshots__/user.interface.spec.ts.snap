// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: User should create an entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "exportOrder": 11,
      "filterName": "_id",
      "forceExport": true,
      "name": "_id",
      "title": "_ID",
      "type": "text",
      "visible": false,
    },
    Object {
      "allowLibrary": true,
      "columnDefinition": Object {
        "width": 52,
      },
      "exportOrder": 10,
      "filterName": "imageData",
      "filterable": false,
      "name": "imageData",
      "sortable": false,
      "title": "PHOTO",
      "type": "photo",
    },
    Object {
      "asyncValidators": Array [
        [Function],
      ],
      "condition": "not (getAttributeValue(\\"_id\\") == \\"multiple\\")",
      "disableAutocomplete": true,
      "exportOrder": 1,
      "filterName": "username",
      "flex": 100,
      "name": "username",
      "readonly": "getAttributeValue(\\"_ect\\")",
      "required": true,
      "title": "USERNAME",
      "type": "email",
    },
    Object {
      "disableAutocomplete": true,
      "exportOrder": 6,
      "filterName": "email",
      "flex": 100,
      "name": "email",
      "required": false,
      "title": "EMAIL",
      "type": "email",
    },
    Object {
      "condition": "not getAttributeValue(\\"_ect\\")",
      "disableAutocomplete": true,
      "exportOrder": 2,
      "filterName": "password",
      "filterable": false,
      "name": "password",
      "required": true,
      "sortable": false,
      "title": "PASSWORD",
      "type": "password",
    },
    Object {
      "exportOrder": 4,
      "filterName": "firstName",
      "flex": 100,
      "name": "firstName",
      "required": true,
      "title": "FIRSTNAME",
      "type": "text",
    },
    Object {
      "exportOrder": 5,
      "filterName": "lastName",
      "flex": 100,
      "name": "lastName",
      "required": true,
      "title": "LASTNAME",
      "type": "text",
    },
    Object {
      "collectionName": "tenants",
      "columnDefinition": Object {
        "name": "name",
      },
      "condition": Array [
        Object {
          "group": Array [
            "admin",
          ],
          "operator": "in",
          "type": "roles",
        },
      ],
      "filterName": "_tenantRef",
      "multiple": false,
      "name": "_tenant",
      "required": true,
      "title": "TENANT",
      "type": "autocomplete",
    },
    Object {
      "condition": Array [
        "not getAttributeValue(\\"_ect\\")",
        Object {
          "group": Array [
            "admin",
          ],
          "operator": "in",
          "type": "roles",
        },
      ],
      "exportOrder": 12,
      "filterName": "isTeam",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "isTeam",
      "title": "TEAM",
      "type": "toggle",
    },
    Object {
      "collectionName": "user",
      "condition": Object {
        "group": Array [
          "admin",
        ],
        "operator": "in",
        "type": "roles",
      },
      "exportOrder": 13,
      "filterName": "tags",
      "icon": "yo-flag",
      "multiple": true,
      "name": "tags",
      "subQuery": Object {
        "field": "ownerRef",
        "values": "_id",
      },
      "tag": true,
      "title": "USERTAGS",
      "type": "autocomplete",
    },
    Object {
      "condition": Array [
        Object {
          "group": Array [
            "admin",
          ],
          "operator": "in",
          "type": "roles",
        },
      ],
      "exportOrder": 15,
      "filterName": "role",
      "filterableAdvanced": true,
      "flex": 100,
      "name": "role",
      "required": true,
      "title": "ROLE",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "ROLEADMIN",
        "ROLEEDITOR",
        "ROLEVIEWER",
        "ROLEMANAGER",
        "ROLEFIELD",
        "ROLESTOREMANAGER",
        "ROLESTORE",
      ],
    },
    Object {
      "advanced": true,
      "exportOrder": 7,
      "filterName": "telephone",
      "filterableAdvanced": true,
      "flex": 50,
      "name": "telephone",
      "title": "TELEPHONE",
      "type": "tel",
    },
    Object {
      "advanced": true,
      "exportOrder": 8,
      "filterName": "company",
      "filterableAdvanced": true,
      "flex": 50,
      "name": "company",
      "title": "COMPANY",
      "type": "text",
    },
    Object {
      "advanced": true,
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "dateOfBirth",
      "filterableAdvanced": true,
      "flex": 50,
      "name": "dateOfBirth",
      "suppressExport": true,
      "title": "DATEOFBIRTH",
      "type": "date",
    },
    Object {
      "advanced": true,
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "gender",
      "filterableAdvanced": true,
      "flex": 50,
      "name": "gender",
      "suppressExport": true,
      "title": "GENDER",
      "translate": true,
      "type": "autocomplete",
      "values": Array [
        "MALE",
        "FEMALE",
      ],
    },
    Object {
      "advanced": true,
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 9,
      "filterName": "address",
      "name": "address",
      "title": "ADDRESS",
      "type": "address",
    },
    Object {
      "clearable": true,
      "collectionName": "locations",
      "condition": Array [
        Object {
          "group": Array [
            "manager",
          ],
          "operator": "in",
          "type": "roles",
        },
        Object {
          "group": Array [
            "trial",
          ],
          "operator": "nin",
          "type": "roles",
        },
        "not (username == \\"smartin@yoobic.com\\")",
        "not (endsWith(getAttributeValue(\\"username\\"),\\"@yoobic.com\\") == 1 and length(\\"_acl.groups.r\\") == 0)",
      ],
      "deleteOnHidden": false,
      "filterName": "locationRef",
      "name": "location",
      "onChange": [Function],
      "suppressExport": true,
      "title": "LOCATION",
      "type": "autocomplete",
    },
    Object {
      "exportOrder": 14,
      "filterName": "locationRef",
      "filterable": false,
      "forceExport": true,
      "name": "locationRef",
      "title": "LOCATIONREF",
      "type": "text",
      "visible": false,
    },
    Object {
      "allowCustomTag": true,
      "clearable": false,
      "collectionName": "groups",
      "columnDefinition": Object {
        "forceName": true,
        "name": "_acl?.groups?.r",
      },
      "condition": Array [
        Object {
          "group": Array [
            "admin",
            "clientadmin",
          ],
          "operator": "in",
          "type": "roles",
        },
        "not (username == \\"smartin@yoobic.com\\")",
        "not (endsWith(getAttributeValue(\\"username\\"),\\"@yoobic.com\\") == 1 and length(\\"_acl.groups.r\\") == 0)",
        Object {
          "group": Array [
            "trial",
          ],
          "operator": "nin",
          "type": "roles",
        },
      ],
      "deleteOnHidden": false,
      "exportOrder": 3,
      "filterName": "_acl.groups.r",
      "filters": Array [
        Array [
          Object {
            "field": "isRole",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
          Object {
            "field": "type",
            "operator": Object {
              "_id": "nin",
            },
            "value": Array [
              "plan",
            ],
          },
        ],
      ],
      "hiddenFields": Array [
        "isRole",
        "type",
      ],
      "idOnly": true,
      "mapTransform": [Function],
      "multiple": true,
      "name": "_acl.groups.r",
      "required": true,
      "title": "GROUPS",
      "type": "autocomplete",
    },
    Object {
      "advanced": true,
      "filterName": "_lmt",
      "filterableAdvanced": true,
      "name": "_lmt",
      "readonly": true,
      "suppressExport": true,
      "title": "_LMT",
      "type": "date",
    },
    Object {
      "advanced": true,
      "filterName": "version",
      "filterableAdvanced": true,
      "name": "version",
      "readonly": true,
      "suppressExport": false,
      "title": "VERSION",
      "type": "text",
    },
    Object {
      "advanced": true,
      "filterName": "mobileVersion",
      "filterableAdvanced": true,
      "name": "mobileVersion",
      "readonly": true,
      "suppressExport": false,
      "title": "MOBILEVERSION",
      "type": "text",
    },
    Object {
      "advanced": true,
      "filterName": "platform",
      "filterableAdvanced": true,
      "name": "platform",
      "readonly": true,
      "suppressExport": true,
      "title": "PLATFORM",
      "type": "text",
    },
    Object {
      "advanced": true,
      "filterName": "language",
      "filterableAdvanced": true,
      "name": "language",
      "readonly": true,
      "suppressExport": true,
      "title": "LANGUAGE",
      "type": "text",
    },
    Object {
      "advanced": true,
      "filterName": "device",
      "filterable": false,
      "name": "device",
      "readonly": true,
      "suppressExport": true,
      "title": "DEVICE",
      "type": "text",
    },
    Object {
      "advanced": true,
      "filterName": "uuid",
      "filterable": false,
      "name": "uuid",
      "readonly": true,
      "suppressExport": true,
      "title": "UUID",
      "type": "text",
    },
    Object {
      "advanced": true,
      "filterName": "target",
      "filterable": false,
      "name": "target",
      "readonly": true,
      "suppressExport": true,
      "title": "TARGET",
      "type": "text",
    },
    Object {
      "condition": Object {
        "group": Array [
          "admin",
        ],
        "operator": "in",
        "type": "roles",
      },
      "filterName": "photoMaxWidth",
      "filterable": false,
      "name": "photoMaxWidth",
      "suppressExport": true,
      "title": "MAXWIDTH",
      "type": "number",
    },
  ],
  "appareance": Map {},
  "className": "User",
  "collectionName": "user",
  "feathersService": undefined,
  "fields": Array [
    "*",
  ],
  "include": Array [
    "location",
    "_tenant",
  ],
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [
    "username",
    "email",
    "firstName",
    "lastName",
  ],
  "type": [Function],
}
`;
